mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping

```diff
@@ -1,32 +1,164 @@
-		METHOD method_32619 value (Ljava/lang/Object;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Builds a cycling button widget.
+			COMMENT
+			COMMENT @throws IllegalStateException if no {@code values} call is made, or the
+			COMMENT {@code values} has no default values available
+			ARG 1 x
+			ARG 2 y
+			ARG 3 width
+			ARG 4 height
+			ARG 5 optionText
+			ARG 6 callback
+		METHOD method_32618 tooltip (Lnet/minecraft/class_5676$class_5679;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the tooltip factory that provides tooltips for any of the values.
+			COMMENT
+			COMMENT <p>If this is not called, the values simply won't have tooltips.
+			ARG 1 tooltipFactory
+		METHOD method_32619 initially (Ljava/lang/Object;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the initial value of this button widget.
+			COMMENT
+			COMMENT <p>This is not effective if {@code value} is not in the default
+			COMMENT values (i.e. excluding alternative values).
+			COMMENT
+			COMMENT <p>If this is not called, the initial value defaults to the first
+			COMMENT value in the values list supplied.
 			ARG 1 value
+		METHOD method_32620 values (Ljava/util/List;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the option values for this button.
+			ARG 1 values
+		METHOD method_32621 values (Ljava/util/List;Ljava/util/List;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the option values for this button.
+			COMMENT
+			COMMENT <p>When the user presses the ALT key, the {@code alternatives} values
+			COMMENT will be iterated; otherwise the {@code defaults} values will be iterated
+			COMMENT when clicking the built button.
+			ARG 1 defaults
+			ARG 2 alternatives
+		METHOD method_32622 values (Ljava/util/function/BooleanSupplier;Ljava/util/List;Ljava/util/List;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the option values for this button.
+			COMMENT
+			COMMENT <p>When {@code alternativeToggle} {@linkplain BooleanSupplier#getAsBoolean()
+			COMMENT getAsBoolean} returns {@code true}, the {@code alternatives} values
+			COMMENT will be iterated; otherwise the {@code defaults} values will be iterated
+			COMMENT when clicking the built button.
+			ARG 1 alternativeToggle
+			ARG 2 defaults
+			ARG 3 alternatives
+		METHOD method_32623 narration (Ljava/util/function/Function;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Overrides the narration message of the button to build.
+			COMMENT
+			COMMENT <p>If this is not called, the button will use
+			COMMENT {@link CyclingButtonWidget#getGenericNarrationMessage()} for narration
+			COMMENT messages.
+			ARG 1 narrationMessageFactory
+		METHOD method_32624 values ([Ljava/lang/Object;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the option values for this button.
+			ARG 1 values
 		METHOD method_32625 (Ljava/lang/Object;)Ljava/util/List;
 			ARG 0 value
-	CLASS class_5678
+	CLASS class_5678 Callback
```

Comment:
```suggestion
	CLASS class_5678 ValueChangeListener
```
Calling it `Callback` is too generic IMO.

Reply:
imo callback suffices as callbacks are for updates, which is what this thing does

Reply:
```suggestion
	CLASS class_5678 UpdateCallback
```